components/product/list.js

```diff
@@ -1,15 +1,45 @@
 import { SingleProductComponent } from "./single";
-
-export const ListProductComponent = ({ products = [], handleAddToCartClick }) => {
+import React, { useState } from "react";
+export const ListProductComponent = ({
+  products = [],
+  handleAddToCartClick,
+}) => {
+  const [sortProduct, setSortProduct] = useState(false);
+  const sortByPrice = () => {
+    setSortProduct(!sortProduct);
+  };
+  const compare = (a, b) => {
+    return a.price - b.price;
+  };
+  const sortedProductByPrice = () => {
+    return products.sort(compare);
+  };
+  console.log(products);
   return (
-    <div className="flex">
-      {products.map((product) => (
-        <SingleProductComponent
-          key={product.name}
-          product={product}
-          onAddToCartClick={handleAddToCartClick}
-        />
-      ))}
-    </div>
+    <>
+      <button
+        className="px-3 py-2 bg-indigo-200 text-gray-800"
+        onClick={sortByPrice}
+      >
+        Sort By
+      </button>
+      <div className="flex">
+        {!!sortProduct
+          ? sortedProductByPrice().map((product) => (
```

Comment:
this works, but it can be done in a much simpler and compact way. Before the component renders, you could do `const productList = !!sortProduct ? sortedProductByPrice() : products;` and then in this line, instead of doing it conditionally, you could just do `productList.map` and that would have the same effect.

Reply:
Thanks for your suggestions vaiya. Next time I'll try it out definitely.